x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/SpanFlyout/index.tsx

```diff
@@ -15,27 +15,74 @@ import {
   EuiPortal,
   EuiSpacer,
   EuiTabbedContent,
-  EuiTitle
+  EuiTitle,
+  EuiBadge,
+  EuiToolTip
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { get, keys } from 'lodash';
 import React, { Fragment } from 'react';
 import styled from 'styled-components';
 import { idx } from '@kbn/elastic-idx';
+import { px, units } from '../../../../../../../style/variables';
+import { Summary } from '../../../../../../shared/Summary';
+import { TimestampSummaryItem } from '../../../../../../shared/Summary/TimestampSummaryItem';
+import { DurationSummaryItem } from '../../../../../../shared/Summary/DurationSummaryItem';
 import { Span } from '../../../../../../../../typings/es_schemas/ui/Span';
 import { Transaction } from '../../../../../../../../typings/es_schemas/ui/Transaction';
 import { DiscoverSpanLink } from '../../../../../../shared/Links/DiscoverLinks/DiscoverSpanLink';
 import { Stacktrace } from '../../../../../../shared/Stacktrace';
-import { FlyoutTopLevelProperties } from '../FlyoutTopLevelProperties';
 import { ResponsiveFlyout } from '../ResponsiveFlyout';
 import { DatabaseContext } from './DatabaseContext';
 import { HttpContext } from './HttpContext';
 import { StickySpanProperties } from './StickySpanProperties';
 
+function formatType(type: string) {
+  switch (type) {
+    case 'db':
+      return 'DB';
+    case 'hard-navigation':
+      return i18n.translate(
+        'xpack.apm.transactionDetails.spanFlyout.spanType.navigationTimingLabel',
+        {
+          defaultMessage: 'Navigation timing'
+        }
+      );
+    default:
+      return type;
+  }
+}
+
+function formatSubtype(subtype: string) {
+  switch (subtype) {
+    case 'mysql':
+      return 'MySQL';
+    default:
+      return subtype;
+  }
+}
+
+function getSpanTypes(span: Span) {
+  const { type, subtype, action } = span.span;
+
+  const [primaryType, subtypeFromType, actionFromType] = type.split('.'); // This is to support 6.x data
```

Comment:
I'm trying to recall if this is still necessary. In 6.x we used to only have the `type` field with dots as delimiter. But to move to 7.x the user had to migrate their data afair. So maybe this is no longer necessary? If you remove it you might want to double check with apm server team.

Reply:
Checked with the server team, it can be removed.